### ISSUE TITLE
"$$slots" variable throws "no-undef" error

### ISSUE THREAD

**Issue by sounisi5011** [sveltejs/eslint-plugin-svelte3#107]
Following [the official documentation](https://svelte.dev/tutorial/optional-slots), I have created the following components for my project.

```svelte
{#if $$slots.labelText}
  <!-- svelte-ignore a11y-label-has-associated-control -->
  <label>
    <slot name="labelText" />
    <slot />
  </label>
{:else}
  <slot />
{/if}
```

This component will compile and run successfully.
However, when I used `eslint-plugin-svelte3@3.1.2`, I got a no-undef error.

```
./LabelInputArea.svelte
  1:6  error  '$$slots' is not defined  no-undef
```

---

**To developers who have encountered similar problems:**

The `$$slots` variable can be used within the script element.
So, I avoided this error by rewriting the following code.

```svelte
<script lang="ts">
  const slots = $$slots;
</script>

{#if slots.labelText}
  <!-- svelte-ignore a11y-label-has-associated-control -->
  <label>
    <slot name="labelText" />
    <slot />
  </label>
{:else}
  <slot />
{/if}
```

**Comment by j3rem1e** [comment]
I did not reproduce this issue with v3.1.2

**Comment by sounisi5011** [comment]
This error occurs in this repository: https://github.com/sounisi5011/web-tool--tiny-template

The code that causes this problem is [`src/components/LabelInputArea.svelte`](https://github.com/sounisi5011/web-tool--tiny-template/blob/fbdc35cd350446d26c36a298736f8b448acf0360/src/components/LabelInputArea.svelte). Currently, I have worked around the error, but if I edit this file to the code I reported and run ESLint, the error recurs.

Later, I will create a small repository where anyone can reproduce this problem.

---

After several attempts, I found out that this error is thrown only when the `script` element is not present in the code.
This error is not thrown by the following code.

```svelte
<script></script>

{#if $$slots.labelText}
{/if}
```

The following code will throw an error.

```svelte
{#if $$slots.labelText}
{/if}
```
